addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -4343,6 +4338,22 @@ const actions = {
         await dispatch('_fetchPartnerImStatus');
         dispatch('_loopFetchPartnerImStatus');
     },
+    /**
+     * @private
+     * @param {Object} param0
+     * @param {Object} param0.state
+     * @param {string} threadCacheLocalId
+     */
+    async _setThreadCacheLoading({ state }, threadCacheLocalId) {
+        const threadCache = state.threadCaches[threadCacheLocalId];
+        threadCache.isLoaded = false;
+        setTimeout(function(){
+            if (!threadCache.isLoaded)
+            {
+                threadCache.isLoading = true;
+            }
+        }, 400);
```

Comment:
I like the idea, but I'm not sure the timeout should be managed in the store.
At the moment I lean toward components displaying "please wait..." in case thread cache is still loading after a certain amount of time.

Also not that you should handle clearing the timeout, otherwise you could have weird glitches like loading.../not loading.../ loading.../ not loading..., if you see what I meant